File: ios/AmphionRuntime/Sources/AmphionRuntime/Internal/Logger.swift
import Foundation

/// 内部 logger 适配器：所有 SDK 内部代码用 `Log.*` 而非直接 print/NSLog，以便统一注入。
internal enum Log {
    static func d(_ msg: @autoclosure () -> String) {
        AsrSdk.shared.currentLogger().debug(msg())
    }
    static func i(_ msg: @autoclosure () -> String) {
        AsrSdk.shared.currentLogger().info(msg())
    }
    static func w(_ msg: @autoclosure () -> String) {
        AsrSdk.shared.currentLogger().warn(msg())
    }
    static func e(_ msg: @autoclosure () -> String, error: Error? = nil) {
        AsrSdk.shared.currentLogger().error(msg(), error: error)
    }
}

File: android/AmphionRuntime/sdk/src/main/java/com/amphion/asr/internal/Logger.kt
package com.amphion.asr.internal

import android.util.Log
import com.amphion.asr.AsrLogLevel

internal object Logger {

    private const val TAG = "AsrSdk"

    @Volatile
    private var minLevel: AsrLogLevel = AsrLogLevel.WARN

    @Volatile
    var httpTimeoutMs: Int = 30_000
        private set

    fun setLevel(level: AsrLogLevel) {
        minLevel = level
    }

    fun setHttpTimeoutMs(ms: Int) {
        httpTimeoutMs = ms
    }

    fun d(msg: String) { if (allow(AsrLogLevel.DEBUG)) Log.d(TAG, msg) }
    fun i(msg: String) { if (allow(AsrLogLevel.INFO))  Log.i(TAG, msg) }
    fun w(msg: String) { if (allow(AsrLogLevel.WARN))  Log.w(TAG, msg) }
    fun e(msg: String, t: Throwable? = null) {
        if (allow(AsrLogLevel.ERROR)) {
            if (t == null) Log.e(TAG, msg) else Log.e(TAG, msg, t)
        }
    }

    private fun allow(level: AsrLogLevel): Boolean {
        if (minLevel == AsrLogLevel.NONE) return false
        return level.ordinal >= minLevel.ordinal
    }
}
